### PR TITLE
[parser] Implement multiline statement

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Related: #298

### CODE REVIEW

File: src/lpython/parser/parser.yy [hunk 1]
```diff
@@ -243,33 +244,21 @@ units
...
 
 script_unit
-    : statement sep   { $$ = SCRIPT_UNIT_STMT($1); }
-    | expr sep        { $$ = SCRIPT_UNIT_EXPR($1); }
-    ;
-/*
-statements
-    : single_stmt_list TK_NEWLINE { }
-    | sep TK_INDENT statements1 TK_DEDENT { }
+    : statement END_OF_FILE
```
Comment (Thirumalai-Shaktivel) [inline src/lpython/parser/parser.yy:247]: I added END_OF_FILE to resolve the following error:
```
syntax error: End of file is unexpected here
  --> examples/expr2.py:11:1
   |
11 | 
   | ^ 
```
This is not the correct approach to fix this, we need to figure out another way.

File: src/lpython/parser/parser.yy [hunk 2]
```diff
@@ -243,33 +244,21 @@ units
...
     ;
 
-statements1
-    : statements1 statement sep { }
-    | statement sep { }
+statements
+    : sep TK_INDENT statement sep TK_DEDENT { $$ = $3; }
```
Comment (Thirumalai-Shaktivel) [inline src/lpython/parser/parser.yy:251]: This line parses only one statement either a single-line statement (or maybe multi-line)
For example:
```python
if True:
    pass
elif True:
    pass
elif True:
    pass
elif True:
    pass
else:
    pass
```

Comment (Thirumalai-Shaktivel) [reply src/lpython/parser/parser.yy:251]: The AST generated for the above example:
```console
$ lpython --show-ast --new-parser --tree examples/expr2.py
└-Module
  |-body=↧
  | └-If
  |   |-test=ConstantBool
  |   | |-value=.true.
  |   | └-kind=()
  |   |-body=↧
  |   | └-Pass
  |   └-orelse=↧
  |     └-If
  |       |-test=ConstantBool
  |       | |-value=.true.
  |       | └-kind=()
  |       |-body=↧
  |       | └-Pass
  |       └-orelse=↧
  |         └-If
  |           |-test=ConstantBool
  |           | |-value=.true.
  |           | └-kind=()
  |           |-body=↧
  |           | └-Pass
  |           └-orelse=↧
  |             └-If
  |               |-test=ConstantBool
  |               | |-value=.true.
  |               | └-kind=()
  |               |-body=↧
  |               | └-Pass
  |               └-orelse=↧
  |                 └-Pass
  └-type_ignores=↧
```
which is identical with the old-parser (CPython)

**Comment by Thirumalai-Shaktivel** [comment]
```python
if True:
    pass
	print()
```
```python
if True:	print()
```

File: src/lpython/parser/semantics.h [hunk 3]
```diff
@@ -165,5 +173,6 @@ int dot_count = 0;
 #define CALL_02(func, args, l) make_Call_t(p.m_a, l, \
         EXPR(func), EXPRS(args), args.size(), nullptr, 0)
 
+#define SCRIPT_UNIT_STMT(x) (ast_t*)x
```
Comment (certik) [inline src/lpython/parser/semantics.h:176]: This is probably not needed.

**Comment by Thirumalai-Shaktivel** [comment]
This PR is ready with `if statements`! 
cc @certik,
Shall I implement other multiline statements or merge this and create separate PR for the same?

**Review by certik — APPROVED** [review]
This looks good, thanks!

**Pull request merged**